plugins/modules/ovirt_host.py

```diff
@@ -610,38 +610,44 @@ def main():
                 )
                 # Set to False, because upgrade_check isn't 'changing' action:
                 hosts_module._changed = False
-            ret = hosts_module.action(
-                action='upgrade',
-                action_condition=lambda h: h.update_available,
-                wait_condition=lambda h: h.status == result_state and (
-                    len([
-                        event
-                        for event in events_service.list(
-                            from_=int(last_event.id),
-                            # Finished upgrade:
-                            # 841: HOST_UPGRADE_FAILED
-                            # 842: HOST_UPGRADE_FINISHED
-                            # 888: HOST_UPGRADE_FINISHED_AND_WILL_BE_REBOOTED
-                            search='type=842 or type=841 or type=888',
-                        ) if host.name in event.description
-                    ]) > 0
-                ),
-                post_action=lambda h: time.sleep(module.params['poll_interval']),
-                fail_condition=lambda h: hosts_module.failed_state_after_reinstall(h) or (
-                    len([
-                        event
-                        for event in events_service.list(
-                            from_=int(last_event.id),
-                            # Fail upgrade if migration fails:
-                            # 17: Failed to switch Host to Maintenance mode
-                            # 65, 140: Migration failed
-                            # 166: No available host was found to migrate VM
-                            search='type=65 or type=140 or type=166 or type=17',
-                        ) if host.name in event.description
-                    ]) > 0
-                ),
-                reboot=module.params['reboot_after_upgrade'],
+            # List all update check events and check if we some updates were found
+            no_upgrade = events_service.list(
+                from_=int(last_event.id),
+                search='type=885 and message="*no updates found*" and host.name=%s' % host.name,
```

Comment:
When there are no updates available, then HostUpgradeCommand failes with specific message:

https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/hostdeploy/UpgradeHostCommand.java#L50
https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/validator/UpgradeHostValidator.java#L45

Wouldn't it be easier to check this result?

Reply:
Was able to find middle ground... now it will skip the wait if the update is not available

Reply:
Great!